Brokerages/Backtesting/BacktestingBrokerage.cs

```diff
@@ -598,5 +596,80 @@ public void ProcessDelistings(Delistings delistings)
                 }
             }
         }
+
+        private void RemoveOrders(List<Order> orders, OrderStatus orderStatus, string message = "")
+        {
+            for (var i = 0; i < orders.Count; i++)
+            {
+                RemoveOrder(orders[i], orderStatus, message);
+            }
+        }
+
+        private void RemoveOrder(Order order, OrderStatus orderStatus, string message = "")
+        {
+            OnOrderEvent(new OrderEvent(order, Algorithm.UtcTime, OrderFee.Zero, message) { Status = orderStatus });
+
+            _pending.TryRemove(order.Id, out var _);
+        }
+
+        private bool TryOrderPreChecks(Dictionary<Order, Security> ordersSecurities, out bool stillNeedsScan)
+        {
+            var result = true;
+            stillNeedsScan = false;
+
+            foreach (var kvp in ordersSecurities)
+            {
+                var order = kvp.Key;
+                var security = kvp.Value;
+
+                if (order.Type == OrderType.MarketOnOpen)
+                {
+                    // This is a performance improvement:
+                    // Since MOO should never fill on the same bar or on stale data (see FillModel)
+                    // the order can remain unfilled for multiple 'scans', so we want to avoid
+                    // margin and portfolio calculations since they are expensive
+                    var currentBar = security.GetLastData();
+                    var localOrderTime = order.Time.ConvertFromUtc(security.Exchange.TimeZone);
+                    if (currentBar == null || localOrderTime >= currentBar.EndTime)
+                    {
+                        stillNeedsScan = true;
+                        result = false;
+                        break;
+                    }
+                }
+
+                // check if the time in force handler allows fills
+                if (order.TimeInForce.IsOrderExpired(security, order))
+                {
+                    RemoveOrder(order, OrderStatus.Canceled, "The order has expired.");
```

Comment:
In this case should remove all of them really? We probably shouldn't use `OnOrderEvent/RemoveOrder` singular anymore here?